src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java

```diff
@@ -711,6 +704,159 @@ public String[] executeSimple(Object[][] params) {
 						"\t\tset {_money} to formatNumber({money::%sender's uuid%})",
 						"\t\tsend \"Your balance: %{_money}%\" to sender")
 			.since("2.10");
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("mean", new Parameter[]{
+			new Parameter<>("numbers", DefaultClasses.NUMBER, false, null)
+		}, DefaultClasses.NUMBER, true) {
+			@Override
+			public Number @Nullable [] executeSimple(Object[][] params) {
+				double total = 0;
+				for (Object object : params[0]) {
+					Number number = (Number) object;
+					total += number.doubleValue();
+				}
+				total /= params[0].length;
+				return new Number[]{total};
+			}
+		})
+			.description("Get the average of a list of numbers.")
+			.examples(
+				"mean(1, 2, 3) = 2",
+				"mean(0, 5, 10) = 5",
+				"mean(13, 97, 376, 709) = 298.75"
+			)
+			.since("INSERT VERSION");
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("median", new Parameter[]{
+			new Parameter<>("numbers", DefaultClasses.NUMBER, false, null)
+		}, DefaultClasses.NUMBER, true) {
+			@Override
+			public Number @Nullable [] executeSimple(Object[][] params) {
+				Number[] sorted = Arrays.stream(params[0])
+					.filter(object -> object instanceof Number)
+					.map(object -> (Number) object)
+					.sorted(((o1, o2) -> {
+						Double n1 = o1.doubleValue();
+						Double n2 = o2.doubleValue();
+						return n1.compareTo(n2);
+					}))
+					.toArray(Number[]::new);
+				int size = sorted.length;
+				if (size % 2 == 1)
+					return new Number[]{sorted[Math2.ceil(size /2)]};
+				int half = size / 2;
+				double first = (sorted[half - 1]).doubleValue();
+				double second = (sorted[half]).doubleValue();
+				double median = (first+second)/2;
+				return new Number[]{median};
+			}
+		})
+			.description("Get the middle value of a sorted list of numbers. If the list has an even number of values, the median is the average of the two middle numbers.")
+			.examples(
+				"median(1, 2, 3, 4, 5) = 3",
+				"median(1, 2, 3, 4, 5, 6) = 3.5",
+				"median(0, 123, 456, 789) = 289.5"
+			)
+			.since("INSERT VERSION");
+
+		Functions.registerFunction(new SimpleJavaFunction<>("factorial", new Parameter[]{
+			new Parameter<>("number", DefaultClasses.NUMBER, true, null)
+		}, DefaultClasses.NUMBER, true) {
+			@Override
+			public Number @Nullable [] executeSimple(Object[][] params) {
+				Number number = (Number) params[0][0];
+				Long total = Math2.factorial(number.longValue());
+				if (total == null)
+					return null;
+				return new Number[]{total};
+			}
+		})
+			.description()
+			.examples()
+			.since("INSERT VERSION");
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("root", new Parameter[]{
```

Comment:
What's the point of this one? We already have a exponent operator:
```
x ^ (1/2)
```

Reply:
A lot of people probably don't know this.

Reply:
I guess that's fair, yeah